kyo-core/shared/src/test/scala/kyo/AsyncTest.scala

```diff
@@ -777,8 +777,8 @@ class AsyncTest extends Test:
                         for
                             _ <- start.release
                             _ <- run.await
-                            _ <- stop.release
                             _ <- result.set(42)
+                            _ <- stop.release
```

Comment:
This test had a race condition and was failing from time to time. It wasn't guaranteed that `result` is set before the other thread is released.